src/lib/parser.c

```diff
@@ -184,6 +184,19 @@ __Z_INLINE parser_error_t parser_printPublicKey(publickey_t *pk,
     return parser_ok;
 }
 
+__Z_INLINE parser_error_t parser_printSignature(raw_signature_t *s,
+                                                char *outVal, uint16_t outValLen,
+                                                uint8_t pageIdx, uint8_t *pageCount) {
+
+    // REVIEW
+    char outBuffer[128];
+    MEMZERO(outBuffer, sizeof(outBuffer));
+
+    parseHexString(outBuffer, s->buffer);
```

Comment:
I don't expect the signature to come as a hexstring but binary so no need to parse it.
The opposite, we need to convert to a hexstring `array_to_hexstr`. Be careful, the parameters are a bit confusing, we should probably need to document this a bit better. `array_to_hexstr(char *dst, const uint8_t *src, uint8_t source_length)` and you need to have at least `source_length*2+1` in the dst array.